xarray/core/pycompat.py

```diff
@@ -31,10 +31,13 @@ def __init__(self, mod):
                 duck_array_type = (duck_array_module.ndarray,)
             elif mod == "sparse":
                 duck_array_type = (duck_array_module.SparseArray,)
+            elif mod == "cubed":
+                duck_array_type = (duck_array_module.CoreArray,)
             else:
                 raise NotImplementedError
 
-        except ImportError:  # pragma: no cover
+        except ImportError as err:  # pragma: no cover
+            print(err)
```

Comment:
When I `import xarray` in ipython locally, in an environment with cubed installed, this print gives

```
dask
<module 'dask' from '/home/tom/miniconda3/envs/cubed/lib/python3.8/site-packages/dask/__init__.py'>
No module named 'sparse'
cannot import name 'is_duck_dask_array' from partially initialized module 'xarray.core.pycompat' (most likely due to a circular import) (/home/tom/Documents/Work/Code/xarray/xarray/core/pycompat.py)
No module named 'cupy'
```

The message `cannot import name 'is_duck_dask_array' from partially initialized module 'xarray.core.pycompat' (most likely due to a circular import) (/home/tom/Documents/Work/Code/xarray/xarray/core/pycompat.py)` is the import of cubed failing somehow. 

I don't really see where the circular reference is, because it's not as if cubed uses `xarray.pycompat.is_duck_dask_array` :/

If I do `importlib.import_module("cubed")` on it's own in ipython then it imports with no problems.

Reply:
Oh wait - trying to import `cubed` in the same environment with xarray installed reveals what the problem is: cubed imports rechunker which imports xarray which (with this PR) attempts to import cubed.

```python
cubed/__init__.py:15: in <module>
    from .array_api import Array
cubed/array_api/__init__.py:3: in <module>
    from .array_object import Array
cubed/array_api/array_object.py:5: in <module>
    from cubed.array_api.creation_functions import asarray
cubed/array_api/creation_functions.py:8: in <module>
    from cubed.core import (
cubed/core/__init__.py:2: in <module>
    from .array import (
cubed/core/array.py:7: in <module>
    from rechunker.executors.python import PythonPipelineExecutor
/home/tom/miniconda3/envs/cubed/lib/python3.8/site-packages/rechunker/__init__.py:7: in <module>
    from .api import Rechunked, rechunk
/home/tom/miniconda3/envs/cubed/lib/python3.8/site-packages/rechunker/api.py:9: in <module>
    import xarray
../xarray/xarray/__init__.py:1: in <module>
    from . import testing, tutorial
../xarray/xarray/testing.py:9: in <module>
    from xarray.core import duck_array_ops, formatting, utils
../xarray/xarray/core/duck_array_ops.py:28: in <module>
    from .pycompat import cupy_array_type, dask_array_type, is_duck_dask_array
../xarray/xarray/core/pycompat.py:59: in <module>
    cub = DuckArrayModule("cubed")
../xarray/xarray/core/pycompat.py:35: in __init__
    duck_array_type = (duck_array_module.CoreArray,)
E   AttributeError: partially initialized module 'cubed' has no attribute 'CoreArray' (most likely due to a circular import)
```

The fix is to only import cubed dynamically in xarray like we already do for pint.